src/nl/hannahsten/texifyidea/refactoring/inlinefile/LatexInlineFileHandler.kt

```diff
@@ -0,0 +1,117 @@
+package nl.hannahsten.texifyidea.refactoring.inlinefile
+
+import com.intellij.lang.Language
+import com.intellij.lang.refactoring.InlineActionHandler
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.psi.PsiElement
+import com.intellij.psi.impl.PsiElementBase
+import com.intellij.psi.util.PsiUtilBase.getElementAtCaret
+import nl.hannahsten.texifyidea.LatexLanguage
+import nl.hannahsten.texifyidea.file.LatexFile
+import nl.hannahsten.texifyidea.psi.impl.LatexCommandsImpl
+import nl.hannahsten.texifyidea.util.Log
+import nl.hannahsten.texifyidea.util.files.isLatexFile
+
+/**
+ * Allows for inlining an include command. Does not do the inlining itself.
+ *
+ * @author jojo2357
+ */
+class LatexInlineFileHandler : InlineActionHandler() {
+
+    override fun isEnabledForLanguage(language: Language?): Boolean {
+        return language == LatexLanguage
+    }
+
+    override fun canInlineElement(element: PsiElement?): Boolean {
+        return canInlineLatexElement(element)
+    }
+
+    override fun inlineElement(project: Project, editor: Editor?, element: PsiElement) {
+        // Resolve the file to be inlined
+        val inlineFile: LatexFile = resolveInlineFile(element) ?: return
+
+        val dialog = LatexInlineFileDialog(
+            project,
+            inlineFile,
+            getReference(element, editor),
+            editor != null
+        )
+
+        if (dialog.myOccurrencesNumber > 0) {
+            if (ApplicationManager.getApplication().isUnitTestMode) {
+                try {
+                    dialog.doAction()
+                }
+                finally {
+                    dialog.close(DialogWrapper.OK_EXIT_CODE, true)
+                }
+            }
+            else {
+                dialog.show()
+            }
+        }
+        else {
+            // TODO No ocurrences, what to do?
+        }
+    }
+}
+
+/**
+ * This is static only so that the unit tests can use it, and also since it can be static
+ */
+fun canInlineLatexElement(element: PsiElement?): Boolean {
+    val out = when (element) {
+        is LatexFile -> {
+            element.containingFile.isLatexFile()
+        }
+
+        is LatexCommandsImpl -> {
+            ((element as LatexCommandsImpl).getName() == "\\input" && (element as PsiElementBase).canNavigateToSource())
```

Comment:
That's because I haven't configured an inspection profile yet for qodana, so I think it's just running a subset of inspections. Please do add a Suppress annotation to make it clear this was intentional

Reply:
Rectified